controllers/imageclusterinstall_controller.go

```diff
@@ -216,15 +225,126 @@ func (r *ImageClusterInstallReconciler) Reconcile(ctx context.Context, req ctrl.
 			return ctrl.Result{}, err
 		}
 
-		if err := r.setClusterInstalledConditions(ctx, ici); err != nil {
-			log.WithError(err).Error("failed to set installed conditions")
+		timedout, err := r.checkClusterTimeout(ctx, log, ici, r.DefaultInstallTimeout)
```

Comment:
We want to set the timeout only in case `clusterDeployment.Spec.Installed != true`.
If the controller already set it to true we don't want to keep querying the spoke (unsure what might trigger reconciliation after the `clusterDeployment.Spec.Installed` is set to `true`).